modules/core/src/parallel.cpp

```diff
@@ -775,26 +796,86 @@ static inline int getNumberOfCPUsImpl()
       }
 
    }
-   return cpusAvailable ? cpusAvailable : 1;
+   return cpusAvailable;
+}
+#endif
+
+#if defined CV_HAVE_CGROUP
+static inline unsigned getNumberOfCPUsCFS(void)
+{
+    std::string file_contents = getFileContents("/sys/fs/cgroup/cpu/cpu.cfs_quota_us");
+    if(file_contents.empty())
+        return 0;
+
+    std::stringstream ss_period(file_contents);
+    int cfs_quota;
+    ss_period >> cfs_quota;
+
+    if(cfs_quota < 1) /* cfs_quota must not be 0 or negative */
+        return 0;
+
+    file_contents = getFileContents("/sys/fs/cgroup/cpu/cpu.cfs_period_us");
+    if(file_contents.empty())
+        return 0;
+
+    std::stringstream ss_quota(file_contents);
+    int cfs_period;
+    ss_quota >> cfs_period;
+
+    if(cfs_period < 1)
+        return 0;
+
+    return (unsigned) cfs_quota/cfs_period;
```

Comment:
This gets taken care of in the return, if 0.

Reply:
I'm about the case with CPU=0.5

Reply:
Oh, I assumed we take floor so we return 0. Should I take ceil?

Reply:
Rounding with truncation is fine except range (0, 1). I believe this should work better:
```
max(1u, (unsigned)cfs_quota/cfs_period);
```

Reply:
Oh ok that makes sense, 0.5 makes 1 the valid count, since we take min_non_zero, we might end up taking a bigger value like 8 from other methods cause this is 0. Thanks for explaining.